### DISCUSSION TITLE
How to show diagrams from github on confluence app.

### DISCUSSION THREAD

**Discussion by marcvn-eng** [jgraph/drawio#3470] [Q&A]
HI,

Apologies if this has been asked, but could not find the answer (although closest is this https://github.com/jgraph/drawio/discussions/3219 which has no answers)

I added draw.io app to Github (enterprise) and to Confluence and individually they work fine. The issue comes in where I am trying to embed a diagram. 

As it stands, the only way I can embed a diagram is by setting the GitHub repo to public and then tick the private box on the Confluence app. This seems odd and the repo is not meant to be public.

We have an Atlassian connection to have GitHub and Jira linked, so assumed that this is enough. Do I need to create a specific Confluence connection?

Tried the raw but that gives a security policy warning so that is also not an option.

Any help/insights would be much appreciated here

**Comment by davidjgraph** [comment]
We've never tested Github enterprise, so wouldn't know the answer.

**Reply by marcvn-eng** [reply]
Thanks David. Are there any plans to look at this in the near future? In meantime I guess I will need to look at alternatives to push it through to confluence

**Reply by davidjgraph** [reply]
No immediate plans, I'm not sure if there's any an issue for it. It's worth adding as an issue, if it's not already there.

**Reply by cathex-sam-holdaway** [reply]
@davidjgraph - Just to clarify, when you hear GitHub Enterprise are you thinking the self-hosted version? That's not what we're using, it's standard github.com with a number of organisations managed under an Enterprise. So to all intents and purposes it should just be a normal GitHub connection using the tab in the `Embed draw.io Diagram` dialog:

![image](https://user-images.githubusercontent.com/96190518/228855747-ade52133-4308-4e31-a3f5-8dfbcec0d023.png)

**Reply by davidjgraph** [reply]
I was thinking self-hosted, yes, thanks for the clarification. Could I please ask you to open a ticket at https://github.com/jgraph/drawio/issues with exact steps and the error(s) you saw?

**Reply by cathex-sam-holdaway** [reply]
Yeah definitely, cheers @davidjgraph

**Reply by cathex-sam-holdaway** [reply]
Oh actually we already have one!

https://github.com/jgraph/drawio/issues/3443

If we could get a bit of input on that response it'd be great, at the moment the general push is away from a paid plugin that doesn't immediately work in a simple way (which is a fair call) and maybe towards something other than draw.io entirely for our diagramming needs. I'd like to keep it as an option so if I can grease the wheels with a bit of progress on the issue I'd be pleased!

**Reply by davidjgraph** [reply]
so you have, this is conf cloud, not conf DC?

**Reply by davidjgraph** [reply]
could you also send your SEN for draw.io to support@draw.io please? We need to link the request to you as paying users do get priority over non-paying.

**Reply by cathex-sam-holdaway** [reply]
> so you have, this is conf cloud, not conf DC?

Yup, that's correct

**Reply by cathex-sam-holdaway** [reply]
> could you also send your SEN for draw.io to [support@draw.io](mailto:support@draw.io) please? We need to link the request to you as paying users do get priority over non-paying.

We're not a paying user yet, the Confluence app was installed on a trial basis to prove its utility
